add test files for upgrader (AOD - single channel, but 2D variable)

shlyaeva opened the pull request:
## Description

https://github.com/JCSDA-internal/ioda/pull/264 adds a new test, testing an upgrader for the case when there is only one "channel" variable, which should be output as 2D variable.

Added file (before upgrade):
```
netcdf upgrader_aod {
dimensions:
	nlocs = 10 ;
	nobs = 10 ;
	nrecs = 10 ;
	nvars = 1 ;
	nchans = 1 ;
variables:
	float frequency@VarMetaData(nchans) ;
	int polarization@VarMetaData(nchans) ;
	float wavenumber@VarMetaData(nchans) ;
	int gsi_use_flag@VarMetaData(nchans) ;
	int sensor_channel@VarMetaData(nchans) ;
	float latitude@MetaData(nlocs) ;
	float longitude@MetaData(nlocs) ;
	float time@MetaData(nlocs) ;
	int surface_type@MetaData(nlocs) ;
	int modis_deep_blue_flag@MetaData(nlocs) ;
	float aerosol_optical_depth_4@ObsValue(nlocs) ;
	float aerosol_optical_depth_4@GsiHofXBc(nlocs) ;
	float aerosol_optical_depth_4@GsiHofX(nlocs) ;
	float aerosol_optical_depth_4@GsiObsError(nlocs) ;
	int aerosol_optical_depth_4@MetaData(nlocs) ;
	int aerosol_optical_depth_4@PreQC(nlocs) ;
	float solar_zenith_angle@MetaData(nlocs) ;
	float solar_azimuth_angle@MetaData(nlocs) ;

// global attributes:
		:date_time = 2018041500 ;
		:satellite = "suomi_npp" ;
		:sensor = "v.viirs-m_npp" ;
		:observation_type = "Aod" ;
}
```

Test reference for the file after the upgrade:
```
netcdf upgrader_aod {
dimensions:
	nchans = 1 ;
	nlocs = 10 ;
	nobs = 10 ;
	nrecs = 10 ;
	nvars = 1 ;
variables:
	int nchans(nchans) ;
		nchans:suggested_chunk_dim = 1LL ;
	float nlocs(nlocs) ;
		nlocs:suggested_chunk_dim = 100LL ;
	float nobs(nobs) ;
		nobs:suggested_chunk_dim = 100LL ;
	float nrecs(nrecs) ;
		nrecs:suggested_chunk_dim = 100LL ;
	float nvars(nvars) ;
		nvars:suggested_chunk_dim = 100LL ;

// global attributes:
		string :_ioda_layout = "ObsGroup" ;
		:_ioda_layout_version = 0 ;
		:date_time = 2018041500 ;
		:satellite = "suomi_npp" ;
		:sensor = "v.viirs-m_npp" ;
		:observation_type = "Aod" ;

group: GsiHofX {
  variables:
  	float aerosol_optical_depth(nlocs, nchans) ;
  } // group GsiHofX

group: GsiHofXBc {
  variables:
  	float aerosol_optical_depth(nlocs, nchans) ;
  } // group GsiHofXBc

group: GsiObsError {
  variables:
  	float aerosol_optical_depth(nlocs, nchans) ;
  } // group GsiObsError

group: MetaData {
  variables:
  	int aerosol_optical_depth_4(nlocs) ;
  	float latitude(nlocs) ;
  	float longitude(nlocs) ;
  	int modis_deep_blue_flag(nlocs) ;
  	float solar_azimuth_angle(nlocs) ;
  	float solar_zenith_angle(nlocs) ;
  	int surface_type(nlocs) ;
  	float time(nlocs) ;
  } // group MetaData

group: ObsValue {
  variables:
  	float aerosol_optical_depth(nlocs, nchans) ;
  } // group ObsValue

group: PreQC {
  variables:
  	int aerosol_optical_depth(nlocs, nchans) ;
  } // group PreQC

group: VarMetaData {
  variables:
  	float frequency(nchans) ;
  	int gsi_use_flag(nchans) ;
  	int polarization(nchans) ;
  	int sensor_channel(nchans) ;
  	float wavenumber(nchans) ;
  } // group VarMetaData
}
```

## Dependencies

Adds test data for:
- [ ] https://github.com/JCSDA-internal/ioda/pull/264
Should be merged before the ioda PR.

srherbener reviewed (approved):
I tested this (with the other companion feature branches) on my iMac and ioda-bundle built successfully and all the tests passed. Thanks for adding this in. It will be nice to manage the test data in git lfs again!

shlyaeva committed:
add test files for upgrader (AOD - single channel, but 2D variable)

wsmigaj reviewed (approved):
Both files look good -- thank you!

shlyaeva committed:
update the test file with a channel-like Metadata variable (has to stay 1D)

shlyaeva commented:
Following discussion and bugfix in https://github.com/JCSDA-internal/ioda/pull/264, I have updated the aod test files with a channel-like MetaData variable `aerosol_optical_depth_4@MetaData` that stays 1D after the upgrader is run. I have updated the description of PR with `ncdump`s of the new files.
